### PR TITLE
Feat/vue routing setup

### LINKED ISSUE #67: Create routes for pages in Vue app
# Create routes for pages in Vue app

## 🚀 Goal
Implement Vue Router in our Vue.js application and set up the following routes, each rendering a specific word:
- `/` (Home page) - renders the word "Home"
- `/about` - renders the word "About"
- `/settings` - renders the word "Settings"

This will create the navigation structure for our Decentralized Web App (DWA) starter in Vue.

## 🤔 Background
We're expanding our DWA starter collection to include Vue.js. This task focuses on setting up the routing system, which is crucial for creating a multi-page application structure in our Vue.js DWA starter.

> This is part of our larger project to create a Vue.js DWA starter. See our [main issue here](https://github.com/TBD54566975/tbd-examples/issues/63) for the full context and list of all related tasks.

## 🔑 Tasks and Acceptance Criteria
- [ ] Install and set up Vue Router in the project
- [ ] Create placeholder components for Home, About, and Settings pages
- [ ] Implement the following routes:
  - `/` for the Home page, rendering the word "Home"
  - `/about` for the About page, rendering the word "About"
  - `/settings` for the Settings page, rendering the word "Settings"
- [ ] Create a navigation menu that allows users to switch between these routes
- [ ] Ensure that the correct component is rendered for each route, displaying the specified word
- [ ] Implement proper error handling for undefined routes (404 page)
- [ ] Ensure the routing system works without any console errors

## 🌟 Resources
- [Vue Router Documentation](https://router.vuejs.org/)
- [Vue.js Documentation](https://vuejs.org/guide/introduction.html)
- [Vue.js Style Guide](https://vuejs.org/style-guide/)

## Getting Started
1. Comment ".take" on this issue to get assigned
2. Fork the repository and create a new branch for this task
3. Follow the tasks outlined above
4. Submit a pull request with your changes
5. Respond to any feedback during the review process

## Questions?
If you have any questions or need clarification, please comment on this issue or join our [Discord](https://github.com/TBD54566975/tbd-examples/issues/64#) community.

Happy coding! 🎉

### PR DESCRIPTION
**Pull request by Johnnyevans32**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 👷 Example Application
- [ ] 🧑‍💻 Code Snippet
- [ ] 🎨 Design
- [ ] 📖 Content
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds the setup for Vue Router in our Vue.js DWA starter project. It implements the routing for the Home, About, and Settings pages and handles undefined routes with a custom 404 page. Each page renders specific content, and a navigation menu is added for switching between routes.

## Related Tickets & Documents

Resolves #67 

## Mobile & Desktop Screenshots/Recordings

https://github.com/user-attachments/assets/4d3b6492-80df-41c4-9ced-8bf62e67de7b

## Added code snippets?
- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 general documentation
- [x] 🙅 no documentation needed

### No docs? Add a note
This is a basic routing setup, and no additional documentation changes are required at this point.

### CODE REVIEW

**Comment by blackgirlbytes** [comment]
@leordev @acekyd pinging you for a second pair of eyes!

**Comment by Johnnyevans32** [comment]
I was wondering if the UI has to be the same as the React counterpart or if it can be different as long as the exact features are implemented. @blackgirlbytes @leordev @finn-tbd

**Comment by acekyd** [comment]
Giving this a look.

File: javascript/dwa-starter-vue/components.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (agpsleo) [inline javascript/dwa-starter-vue/components.json:1]: props on using ShadCN Vue here! 👍

Comment (leordev) [inline javascript/dwa-starter-vue/components.json:1]: Props on using ShadCN Vue! 🚀 🚀

**Comment by leordev** [comment]
> I was wondering if the UI has to be the same as the React counterpart or if it can be different as long as the exact features are implemented. @blackgirlbytes @leordev @finn-tbd

To answer your question, this is an app that will serve as a starter template for any developer in the community to be able to code a DWA, so we want them to have the same uniformed experience regardless of the framework they choose (React, Vue, VanillaJS, Solid, Svelte etc.). So we should in fact have the same components and same UI layout to make it as intuitive as possible for the community.

Could you please style as such? It does not need to be pixel perfect, but same layout composition, fonts and colors are good enough!

**Comment by Johnnyevans32** [comment]
> > I was wondering if the UI has to be the same as the React counterpart or if it can be different as long as the exact features are implemented. @blackgirlbytes @leordev @finn-tbd
> 
> To answer your question, this is an app that will serve as a starter template for any developer in the community to be able to code a DWA, possibly from a CLI tool.
> 
> Thus, we want them to have the same uniformed experience regardless of the user chosen framework (React, Vue, VanillaJS, Solid, Svelte etc.). So, we must have the same components and same UI layout to make it as intuitive as possible for the community.
> 
> Could you please style as such? It does not need to be pixel perfect, but same layout composition, fonts and colors are good enough!

yeah will do, thanks for clearing that out @leordev

**Comment by Johnnyevans32** [comment]
here is the screen recording of the new referenced layout update

https://github.com/user-attachments/assets/2a140633-f0bc-4977-960b-353abfb307c4

**Review by leordev — APPROVED** [review]
Great job @Johnnyevans32 ! 🚀  (merging as soon as the CI retry completes)

**Pull request merged**